Improvements related to editing of features

nprigour opened the pull request:
The FeaturePropertySource & AttributePropertyDescriptor enhancements include:
- fix certain object casts
- send edit commands only in case of actual value change
- handling of empty String values as null
- DateTimeCellEditor added
- NumberCellEditor added that can hnadle setting of null values for numbers 

derived from splitting of #254
 
Signed-off-by: Nikolaos Pringouris <nprigour@gmail.com>

fgdrf commented on plugins/org.locationtech.udig.ui.tests/src/org/locationtech/udig/ui/NumberCellEditorTest.java:
```diff
@@ -0,0 +1,119 @@
+        assertNull(editor.getValue() );
+        
+        //not parsable number
+        editor.setValue("aa");
```
this might be obsolete if `isValueValid`is implemented...

fgdrf commented on plugins/org.locationtech.udig.ui/src/org/locationtech/udig/ui/DateTimeCellEditor.java:
```diff
@@ -0,0 +1,118 @@
+    public void setDateFormatter(DateFormat format) {
+        this.format = format;
+    }
+    
```
Override `isValueValid` here as well to avoid clearance of the feature attribute value in case of ParseException

nprigour commented on plugins/org.locationtech.udig.ui.tests/src/org/locationtech/udig/ui/NumberCellEditorTest.java:
```diff
@@ -0,0 +1,119 @@
+        assertNull(editor.getValue() );
+    }
+    
+    private void runTest( Object value, Object value2, Class<? extends Number> class1 ) {
```
actually for that I just adopted the same naming approach as used for BasicTypeCellEditorTest

nprigour commented:
Hi @fgdrf ,

I have addressed most of your comments/requests with the exception of renaming of runTest (I have explained why.
Moreover in addition to the requested changes I performed:

- some formatting of code and 
- changes in TableView class to allow correct setting of null values during editing of a feature in table view

nprigour commented:
Hi @fgdrf ,

I have addressed most of your comments/requests with the exception of renaming of runTest (I have explained why.
Moreover in addition to the requested changes I performed:

- some formatting of code and 
- changes in TableView class to allow correct setting of null values during editing of a feature in table view

fgdrf commented:
> some formatting of code

I'm wondering about your format settings. Have you configured [codeformatter.xml settings](https://github.com/locationtech/udig-platform/blob/master/extras/org.locationtech.udig.dev/codeformatter.xml) for your workspace (Preferences —> Java —> Code Style —> Code Formatter)?

> exception of renaming of runTest ..

I'm fine with it althought it is not convenient naming such methods. It was just a hint to improve codebase while other classes still might have "special" style ;) And, especially in this case, its copied or used as template which makes it even worse. 

> Besides all the above and before merging the pull request I would like your opinion if it is worthy to 
> perform also some code refactoring to gather for exapmle all cell editors in a separate package
> witihin orgo.locationtech.udig.ui plugin. What do you think?

Good Idea and I'd like to suggest to seperate it from this pull request. We can collaborate on your suggested refacturings (sub-packaging).

nprigour commented:
Yes I use the specified code formatter. However occasionally it may occur that during editing formatting does not apply and sometimesI need to explixitly select format from the eclipse editor context menu.

Ok no objection to handle refactoring to a subpackage in a separate pull request.

fgdrf commented:
Started a build : https://ci.eclipse.org/udig/job/uDig-pullRequest-Build/16/

fgdrf commented:
Can you try to select all lines in at least new files you created and Format code again? Do you have configures some specific save Actions (Preferences -> Java -> Editor -> Save Actions)?

nprigour commented:
re-apply formatting in:

NumberCellEditor.java
NumberCellEditorTest.java
DateTimeCellEditor.java
DateTimeCellEditorTest.java
DateTimePickerDialog.java

nprigour commented:
does this wms related fix conflicts with the present pull? It seems not. Do you want to just merge and push or rebase and force push?

nprigour commented:
rebase succesfully applied

fgdrf commented:
build started : https://ci.eclipse.org/udig/job/uDig-PR/3/
